tools/Microsoft.AspNetCore.Server.Kestrel.GeneratedCode/KnownHeaders.cs

```diff
@@ -31,19 +31,27 @@ static string If(bool condition, Func<string> formatter)
                     case {byLength.Key}:
                         {{{Each(byLength, header => $@"
                             if ({header.EqualIgnoreCaseBytes()})
-                            {{
+                            {{{(header.Identifier == "ContentLength" ? $@"
+                                if (ContentLength.HasValue)
+                                {{
+                                    ThrowInvalid{(className == "FrameResponseHeaders" ? "Response" : "Request")}ContentLengthException(AppendValue(HeaderUtilities.FormatInt64(ContentLength.Value), value).ToString());
```

Comment:
Is this meant to guard against more than one Content-Length header? If so, an error message like "Invalid Content-Length: \\"{value}\\". Value must be a positive integral number." seems misleading.